### PR TITLE
fix(refs dplan-12315): Pass Item to dataTable instead of header-fields.

### PR DESCRIPTION
**Pull request by salisdemos**
coming from DpDataTableExtended, the whole row-item should be available in slots, not the headerfield informations.

### CODE REVIEW

File: src/components/DpDataTable/DpDataTable.vue [hunk 1]
```diff
@@ -484,15 +484,18 @@ export default {
...
+        if (this.isResizable) {
+          this.$nextTick(() => {
+            const firstRow = this.tableEl.getElementsByTagName('tr')[0]
+            const tableHeaderElements = firstRow ? firstRow.children : null
+
+            this.setColsWidth(tableHeaderElements)
+          })
+        }
+      },
+      deep: false // HeaderFields are always replaces as a whole and therefor deep watch is not necessary
```
Comment (spiess-demos) [inline src/components/DpDataTable/DpDataTable.vue:498]: I think this is the default for watchers. See [Vue3 docs](https://vuejs.org/guide/essentials/watchers#deep-watchers).

File: src/components/DpDataTableExtended/DpDataTableExtended.vue [hunk 2]
```diff
@@ -65,24 +65,12 @@
         </slot>
       </template>
       <template
-        v-for="el in filteredFields"
-        v-slot:[el.field]="element">
+        v-for="el in [...filteredFields, { field: 'expandedContent' }, { fields: 'flyout' }]"
```
Comment (spiess-demos) [inline src/components/DpDataTableExtended/DpDataTableExtended.vue:68]: If i correctly get the idea, i think it should be: 
 
```suggestion
        v-for="el in [...filteredFields, { field: 'expandedContent' }, { field: 'flyout' }]"
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: src/components/DpDataTable/DpDataTable.vue (hunk 1 above)
Comment (salisdemos) [reply src/components/DpDataTable/DpDataTable.vue:498]: true. I added it, to make clear, that I checked the spot and its fine and the warning won't point me to that line again and again

**Review by spiess-demos — APPROVED** [review]
lgtm (code review), assuming that this targets the right branch.

**Pull request merged**